corehq/motech/dhis2/management/commands/dhis2_conn_to_conn_settings.py

```diff
@@ -0,0 +1,72 @@
+"""
+Migrate Dhis2Connection instances to ConnectionSettings model
+"""
+from django.core.management import BaseCommand
+
+from corehq.motech.dhis2.dbaccessors import get_dataset_maps
+from corehq.motech.dhis2.models import Dhis2Connection
+from corehq.motech.models import ConnectionSettings
+from corehq.motech.repeaters.models import BASIC_AUTH
+
+
+class Command(BaseCommand):
+
+    def handle(self, **options):
+        for dhis2_conn in Dhis2Connection.objects.all():
+            if conn_settings_exists(dhis2_conn):
+                continue
+            conn_settings = create_conn_settings(dhis2_conn)
+            link_data_set_maps(conn_settings)
+
+
+def conn_settings_exists(dhis2_conn: Dhis2Connection) -> bool:
+    """
+    Does a ConnectionSettings instance exist for ``dhis2_conn``.
+    """
+    return ConnectionSettings.objects\
+        .filter(domain=dhis2_conn.domain)\
+        .filter(url=strip_api(dhis2_conn.server_url))\
+        .count()
+
+
+def strip_api(server_url):
+    """
+    Strips "api/" from the end of ``server_url``, if present
+
+    >>> strip_api('https://play.dhis2.org/demo/api/')
+    'https://play.dhis2.org/demo/'
```

Comment:
This change is to bring DHIS2 aggregated data integration in line with other integrations in HQ/MOTECH. 

Administrators only need to know the base URL of DHIS2, OpenMRS, etc.; We don't need them to know the URL to the API. We can figure that out ourselves, because we know what we are integration with.